modules/dex/src/lib.rs

```diff
@@ -215,6 +385,122 @@ impl<T: Trait> Module<T> {
 
+	/// Access status of specific trading_pair,
+	/// if status is Provisioning and able to be `Enabled`, update it and return
+	/// `Enabled`
+	fn access_trading_pair_status(trading_pair: TradingPair) -> TradingPairStatus<Balance, T::BlockNumber> {
+		let trading_pair_status = Self::trading_pair_statuses(trading_pair);
+		match trading_pair_status {
+			TradingPairStatus::<_, _>::Provisioning(provision_parameters) => {
+				// convert Provisioning to Enabled if possible
+				if frame_system::Module::<T>::block_number() >= provision_parameters.not_before
+					&& !provision_parameters.accumulated_provision.0.is_zero()
+					&& !provision_parameters.accumulated_provision.1.is_zero()
+					&& provision_parameters.accumulated_provision.0 >= provision_parameters.target_provision.0
+					&& provision_parameters.accumulated_provision.1 >= provision_parameters.target_provision.1
+				{
+					let initial_price_0_in_1: Price = Price::checked_from_rational(
+						provision_parameters.accumulated_provision.0,
+						provision_parameters.accumulated_provision.1,
+					)
+					.unwrap_or_default();
+					let lp_share_currency_id = trading_pair.get_dex_share_currency_id().expect("shouldn't be invalid!");
+
+					for (who, contribution) in ProvisioningPool::<T>::drain_prefix(trading_pair) {
+						let share_amount = initial_price_0_in_1
+							.saturating_mul_int(contribution.0)
+							.saturating_add(contribution.1);
+
+						// ignore result to continue
+						let _ = T::Currency::deposit(lp_share_currency_id, &who, share_amount);
+
+						// decrease ref count
+						frame_system::Module::<T>::dec_ref(&who);
+					}
+
+					// inject to liquidity pool
+					LiquidityPool::mutate(trading_pair, |(pool_0, pool_1)| {
+						*pool_0 = pool_0.saturating_add(provision_parameters.accumulated_provision.0);
+						*pool_1 = pool_1.saturating_sub(provision_parameters.accumulated_provision.1);
+					});
+
+					// update trading_pair to Enabled status
+					TradingPairStatuses::<T>::insert(trading_pair, TradingPairStatus::<_, _>::Enabled);
+
+					TradingPairStatus::<_, _>::Enabled
+				} else {
+					trading_pair_status
+				}
+			}
+			_ => trading_pair_status,
+		}
+	}
+
+	/// Add provision to Provisioning TradingPair
+	fn do_add_provision(
+		who: &T::AccountId,
+		currency_id_a: CurrencyId,
+		currency_id_b: CurrencyId,
+		contribution_a: Balance,
+		contribution_b: Balance,
+	) -> DispatchResult {
+		let trading_pair = TradingPair::new(currency_id_a, currency_id_b);
+		let mut provision_parameters = match Self::access_trading_pair_status(trading_pair) {
+			TradingPairStatus::<_, _>::Provisioning(provision_parameters) => provision_parameters,
+			_ => return Err(Error::<T>::MustProvisioningTradingPair.into()),
+		};
+		let (contribution_0, contribution_1) = if currency_id_a == trading_pair.0 {
+			(contribution_a, contribution_b)
+		} else {
+			(contribution_b, contribution_a)
+		};
+
+		ensure!(
+			contribution_0 >= provision_parameters.min_contribution.0
```

Comment:
one of the pair above limit is enough. so people can only add aUSD which is basically market buy order